### DISCUSSION TITLE
babel/polyfill adds global variable core

### DISCUSSION THREAD

**Discussion by bostandyksoft** [babel/babel#13267] [Show and tell]
I have tried to create a small adhoc library. It should be used in our (also interal) another projects. Some projects should work with IE9+.

So i've attached small example of my approach to do this.
[babel_test.zip](https://github.com/babel/babel/files/6431260/babel_test.zip)

For supporting IE9+ within my library i tried to use babel/polyfill - it replaces Promises, that i've used to some polyfill. 

But one of our project already defines global variable "core" for own usage. And when my library is loaded it replaces "core" to babel/polyfills "core". How i can avoid replacing global variable "core"?

![image](https://user-images.githubusercontent.com/4920442/117230909-4c8b2d80-ae48-11eb-80d1-4cf1c3600ead.png)

Best regards. Alexey

**Comment by zloirock** [comment]
`babel-polyfill` is deprecated, use the actual version of `core-js` - in the actual versions the global `core` variable is no longer created.

**Reply by bostandyksoft** [reply]
Hi. I tried to exclude babel-polyfill (as suggest webpack), but could not find any works variant. Can you please provide sample code (or link to it) with using core-js

Best regards

**Reply by zloirock** [reply]
Depends on how do you use `babel-polyfill`. In most cases, just replace import of `babel-polyfill` to `core-js/stable`.

**Reply by bostandyksoft** [reply]
There is (original message) attached sample project

index.js
```
require('@babel/polyfill')

module.exports = {
    calculate : function(a, b) {
        return new Promise(resolve => {
            resolve(a+b);
        })
    }
}
```

.babelrc
```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": false,
        corejs: '3.7.0',
        "useBuiltIns" : "entry"
      }
    ]
  ]
}
```

**Reply by bostandyksoft** [reply]
And require('core-js/stable') did not works

**Reply by nicolo-ribaudo** [reply]
@bostandyksoft What did you mean exactly by "did not work"? It throws an error? It does nothing? It does something else?

**Reply by bostandyksoft** [reply]
Yes. I have got many errors like folowing
![image](https://user-images.githubusercontent.com/4920442/117524065-08ce2a80-afe6-11eb-82a8-ab2537494b12.png)

**Reply by nicolo-ribaudo** [reply]
Does `D:\babel_test\node_modules\core-js\modules\es.symbol.js` exist? If it does, this seems like a webpack (or a webpack config) bug :thinking:

**Reply by bostandyksoft** [reply]
There is no es.symbol.js (etc). Just es6.*, es7.* modules. Should i install some additional package more over core-js?

```
"devDependencies": {
    "@babel/polyfill": "^7.12.1",
    "@babel/preset-env": "^7.14.1",
    "babel-loader": "^8.2.2",
    "clean-webpack-plugin": "^4.0.0-alpha.0",
    "copy-webpack-plugin": "^8.1.1",
    "core-js": "^2.6.12",
    "html-webpack-plugin": "^5.3.1",
    "webpack": "^5.36.2",
    "webpack-cli": "^4.6.0"
  }
```

**Reply by zloirock** [reply]
> `corejs: '3.7.0',`
> `"core-js": "^2.6.12",`
